### PR TITLE
QoL Data Export

### PR DESCRIPTION
**Pull request by Mapleia**
# Changes
Added an endpoint for QoL survey.

## Requirements
Requires:
N/A

## PR Dependencies
Depends on: #13  

## Merged Branches and Other Dependencies
Merged branch(es): `model-schema-change`
Other dependencies:
N/A

## Subtasks / Progress
- [x] Implement survey endpoint (now under measurement but with named source)
- [x] Implement db schema 

## Testing
Unknown as of yet.

- [x] Automated

List of added tests:
-

### CODE REVIEW

File: services/http/api/lib/interface/index.js [hunk 1]
```diff
@@ -33,4 +33,13 @@ module.exports = {
             {...m, time, day, month, year })
         return user.records.length > 0
     },
+
+    createSurvey: (s) => async (db) => {
+        const date = new Date(s.date * 1000)
+        const year = date.getFullYear()
+        const month = date.getMonth()
+        const day = date.getDate()
+        const time = 3600*date.getHours() + 60*date.getMinutes() + date.getSeconds()
```
Comment (rlnsy) [inline services/http/api/lib/interface/index.js]: Can we abstract this to a util function and use it in the function above?

File: services/db_shell/fixtures/test_data.cypher [hunk 2]
```diff
@@ -15,6 +17,12 @@ MATCH (i:Identity{name: "test_export_key"})
 MATCH (i:Identity{name: "test_export_key"})
     MATCH (r:Resource{path: "/measurement"})
     CREATE (i)-[:Can{method: "POST"}]->(r);
+MATCH (i:Identity{name: "test_export_key"})
+    MATCH (r:Resource{path: "/survey"})
+    CREATE (i)-[:Can{method: "POST"}]->(r);
+MATCH (i:Identity{name: "test_export_key"})
+    MATCH (r:Resource{path: "/"})
+    CREATE (i)-[:Can{method: "GET"}]->(r);
```
Comment (rlnsy) [inline services/db_shell/fixtures/test_data.cypher]: Why do we need to allow GET for export?

File: services/http/api/main.js [hunk 3]
```diff
@@ -78,6 +77,16 @@ app.post('/measurement', async (req, res) => {
...
+app.post('/survey', (req, res) => {
+    res.status(200).send({
+        message: 'Received'
+    })
+});
+
+app.get('/', (req, res) => {
+    return res.status(200).end();
+});
+
```
Comment (rlnsy) [inline services/http/api/main.js]: We should add specification for these endpoints in the API reference. I can tell you more about this in a meeting.

**Review by rlnsy — COMMENTED** [review]
This is a great start! Sufficient for testing the firebase app for now. Of course, well want to expand the implementation before merging. See my comments.

File: services/http/api/lib/interface/index.js (hunk 1 above)
Comment (Mapleia) [reply services/http/api/lib/interface/index.js]: Sounds like a good idea. 

```.js
function dateTransformer(input) {
    const date = new Date(input * 1000);
    const year = date.getFullYear();
    const month = date.getMonth();
    const day = date.getDate();
    const time = 3600 * date.getHours() + 60 * date.getMinutes() + date.getSeconds();
    
    return {
        year, 
        month,
        day, 
        time
    }
}

```

This was added!

File: services/db_shell/fixtures/test_data.cypher (hunk 2 above)
Comment (Mapleia) [reply services/db_shell/fixtures/test_data.cypher]: Oh sorry this was to test some things about the apikey, (mostly though Postman). It's just doing GET on the homepage, the `http://localhost:PORT/` page.

File: services/http/api/main.js (hunk 3 above)
Comment (Mapleia) [reply services/http/api/main.js]: Sounds like a great idea! When would be available to having this meeting?

Edit: Should further development of the schema be added after a merge of the OGM branch is done?

**Pull request merged**